packages/node_modules/@node-red/nodes/core/logic/10-switch.html

```diff
@@ -226,50 +245,65 @@
                             {value:"undefined",label:"undefined",hasValue:false},
                             {value:"null",label:"null",hasValue:false}
                         ]});
+                    }
+
+                    var valueField = null;
+                    var numValueField = null;
+                    var expValueField = null;
+                    var btwnAndLabel = null;
+                    var btwnValue2Field = null;
+                    var typeValueField = null;
+                        
                     var finalspan = $('<span/>',{style:"float: right;margin-top: 6px;"}).appendTo(row);
                     finalspan.append(' &#8594; <span class="node-input-rule-index">'+(i+1)+'</span> ');
                     var caseSensitive = $('<input/>',{id:"node-input-rule-case-"+i,class:"node-input-rule-case",type:"checkbox",style:"width:auto;vertical-align:top"}).appendTo(row2);
                     $('<label/>',{for:"node-input-rule-case-"+i,style:"margin-left: 3px;"}).text(caseLabel).appendTo(row2);
                     selectField.change(function() {
                         resizeRule(container);
                         var type = selectField.val();
-                        if ((type === "btwn") || (type === "index")) {
+                        
+                        try {
```

Comment:
Rather than wrap each of these calls in a try/catch, it would probably be cheaper to wrap in a check for the variable existing:
```
if (valueField) {
   valueField.typedInput('hide');
}
```

Reply:
Thanks Nick, I was thinking about that on the way home the other day :)